apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -179,6 +173,53 @@ pub fn validate(schema: Schema) -> Vec<Message> {
     messages
 }
 
+fn check_conflicting_directives(schema: &Schema) -> Vec<Message> {
+    let Some((fed_link, fed_link_directive)) =
+        Link::for_identity(schema, &federation_link_identity())
+    else {
+        return Vec::new();
+    };
+
+    // TODO: make the `Link` code retain locations directly instead of reparsing stuff for validation
+    let imports = fed_link_directive
+        .argument_by_name(&name!("import"))
+        .and_then(|arg| arg.as_list())
+        .into_iter()
+        .flatten()
+        .filter_map(|value| Import::from_value(value).ok().map(|import| (value, import)))
+        .collect_vec();
+
+    let disallowed_imports = [
+        INTF_OBJECT_DIRECTIVE_NAME,
+        CONTEXT_DIRECTIVE_NAME,
+        FROM_CONTEXT_DIRECTIVE_NAME,
+    ];
+    fed_link
+        .imports
+        .into_iter()
+        .filter_map(|import| {
+            disallowed_imports
+                .contains(&import.element)
+                .then(|| Message {
+                    code: Code::UnsupportedFederationDirective,
+                    message: format!(
+                        "The directive `@{import}` is not supported when using connectors.",
+                        import = import.alias.as_ref().unwrap_or(&import.element)
+                    ),
```

Comment:
TODO: update messaging to link to docs to provide more context using our short links one docs exist

[see thread](https://apollograph.slack.com/archives/C0721M2F6/p1719943994783629)